bsp/nrf/lh2.c

```diff
@@ -1000,20 +1605,84 @@ void _spi_setup(const gpio_t *gpio_d) {
     NRF_SPIM->ENABLE = SPIM_ENABLE_ENABLE_Enabled << SPIM_ENABLE_ENABLE_Pos;
 }
 
+void _init_spi_ring_buffer(lh2_ring_buffer_t *cb) {
+    cb->writeIndex = 0;
+    cb->readIndex  = 0;
+    cb->count      = 0;
+
+    for (uint8_t i = 0; i < LH2_BUFFER_SIZE; i++) {
+        memset(cb->buffer[i], 0, SPI_BUFFER_SIZE);
+    }
+}
+
+void _add_to_spi_ring_buffer(lh2_ring_buffer_t *cb, uint8_t data[SPI_BUFFER_SIZE], uint32_t timestamp) {
+
+    memcpy(cb->buffer[cb->writeIndex], data, SPI_BUFFER_SIZE);
+    cb->timestamps[cb->writeIndex] = timestamp;
+    cb->writeIndex                 = (cb->writeIndex + 1) % LH2_BUFFER_SIZE;
+
+    if (cb->count < LH2_BUFFER_SIZE) {
+        cb->count++;
+    } else {
+        // Overwrite oldest data, adjust readIndex
+        cb->readIndex = (cb->readIndex + 1) % LH2_BUFFER_SIZE;
+    }
+}
+
+bool _get_from_spi_ring_buffer(lh2_ring_buffer_t *cb, uint8_t data[SPI_BUFFER_SIZE], uint32_t *timestamp) {
```

Comment:
The signature is weird, I usually do:

```suggestion
bool _get_from_spi_ring_buffer(lh2_ring_buffer_t *cb, uint8_t *data, uint32_t *timestamp) {
```

Reply:
fixed